utils/api/api.go

```diff
@@ -58,3 +58,23 @@ func CheckAPICallSuccess(ctx context.Context, errorContext string, resp *_http.R
 	}
 	return
 }
+
+// FetchAndCheckSuccess is a wrapper for making an API call and then checking success with `CheckAPICallSuccess`
+// errorContext corresponds to the description of what led to the error if error there is e.g. `Failed adding a user`.
+// apiCallFunc corresponds to a generic function that will be called to make the API call
+func FetchAndCheckSuccess[T any](ctx context.Context, errorContext string, apiCallFunc func(ctx context.Context) (T, *_http.Response, error)) (result T, err error) {
```

Comment:
```suggestion
func CallAndCheckSuccess[T any](ctx context.Context, errorContext string, apiCallFunc func(ctx context.Context) (T, *_http.Response, error)) (result T, err error) {
```

could it be called CallAndCheckSuccess because it could be any http function and not just gets